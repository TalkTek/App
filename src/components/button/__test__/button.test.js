import React from 'react'
import renderer from 'react-test-renderer'
import { Button } from '../index'
import { shallow } from 'enzyme'

describe('<Button />', () => {
  it('should render <Button /> elemnet without error', () => {
    shallow(<Button />).dive()
  })

  const testProps = {
    border: 1,
    text: 'just a text',
    borderRadius: 5,
    textColor: 'black'
  }

  const outPutStyle = [
    { 
      borderColor: 'rgb(158, 158, 158)',
      borderRadius: 5,
      justifyContent: 'center',
      alignItems: 'center',
      borderWidth: 1
    },
      undefined 
  ]

  it('should render a correct style button', () => {
    const wrapper = shallow(<Button {...testProps} />).dive()
    expect(wrapper.prop('style')).toEqual(outPutStyle)
  })

  test('render output should be same with expect', () => {
    const tree = renderer.create(<Button {...testProps}/>)
    expect(tree).toMatchSnapshot()
  })

})