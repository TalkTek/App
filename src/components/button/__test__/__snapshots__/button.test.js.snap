// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> render output should be same with expect 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": undefined,
      "borderColor": "rgb(158, 158, 158)",
      "borderRadius": 5,
      "borderWidth": 1,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      black={true}
      ellipsizeMode="tail"
      fontSize={15}
      gray={null}
      green={null}
      style={
        Array [
          Object {
            "color": "rgb(33, 33, 33)",
            "fontSize": 15,
            "fontWeight": null,
          },
          undefined,
        ]
      }
      white={null}
    >
      just a text
    </Text>
  </View>
</View>
`;
